### PR TITLE
pygit2 1.11.1 (new formula) + dependencies

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

We currently bundle `pygit2` as a resource with multiple formulae. However,
this causes difficulty when updating `libgit2`, because we often have to
rebuild all those formulae with some `libgit2` updates.

Let's simplify maintenance of those formulae by adding `pygit2` as a
formula instead. We can then have those formulae depend on `pygit2` to
minimise the required rebuilds from updates of `libgit2`.

In order to package `pygit2`, we also need to package its dependencies,
`pycparser` and `cffi`.

- pycparser 2.21 (new formula)
- cffi 1.15.1 (new formula)
- pygit2 1.11.1 (new formula)

### CODE REVIEW

**Comment by carlocab** [comment]
Added context from https://github.com/Homebrew/homebrew-core/pull/124227#issuecomment-1449545697:

> In order to avoid having to rebuild multiple formulae that use a `pygit2` resource, I'm adding a `pygit2` formula in #124511. This should save a lot of time here, given that it'll allow us to get rid of a `dvc` rebuild (which is a `long build` label all by itself).

**Comment by SMillerDev** [comment]
```
  ==> python3.11 -c 'import _sum_cffi; print(_sum_cffi.lib.sum(1, 2))'
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
  ImportError: libsum.so: cannot open shared object file: No such file or directory
```

File: Formula/cffi.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        int sum(int a, int b);
+      """
+
+      ffibuilder.cdef(declaration)
+      ffibuilder.set_source("_sum_cffi", declaration, libraries=['sum'])
+
+      ffibuilder.compile(verbose=True)
+    EOS
+
+    ENV.append "LDFLAGS", "-L#{testpath} -rpath #{testpath}"
```
Comment (carlocab) [inline Formula/cffi.rb:42]:
```suggestion
    ENV.append "LDFLAGS", "-L#{testpath} -Wl,-rpath,#{testpath}"
```

**Review by SMillerDev — DISMISSED** [review]

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
```
time="2023-03-01T13:46:09Z" level=fatal msg="Error parsing image name \"docker://ghcr.io/homebrew/core/cffi:1.15.1\": reading manifest 1.15.1 in ghcr.io/homebrew/core/cffi: received unexpected HTTP status: 500 Internal Server Error"
```

**Review by ZhongRuoyu — DISMISSED** [review]

**Comment by ZhongRuoyu** [comment]
Whoops, forgot to add `--no-autosquash`.

**Review by ZhongRuoyu — DISMISSED** [review]

**Comment by carlocab** [comment]
@SMillerDev can delete the uploaded package and we can dispatch another publish job here.

**Comment by carlocab** [comment]
I forgot we can pass `--warn-on-upload-failure` to the `pr-pull` call that the workflow does to ignore the fact that `pygit2` is already there. It's a little risky since it also means that other upload failures will be ignored (i.e. for `pycparser` or `cffi` too), but it worked here.

**Pull request closed without merging**